### ISSUE TITLE
Seperate out memory allocation and actual init of device

### ISSUE THREAD

**Issue by EUdds** [bennyE31/pod-plus-plus-test#1]
https://github.com/bennyE31/pod-plus-plus-test/blob/5ddc7ea6f14d13ddf696588aefd8551f4981a91b/i2c.cpp#L5

Generally, it's best practice to separate out the construction of a variable/allocation of memory and the actual "beginning" of the device by sending some data to it over a serial protocol. Not super major just something to note

**Comment by bennyE31** [comment]
> https://github.com/bennyE31/pod-plus-plus-test/blob/5ddc7ea6f14d13ddf696588aefd8551f4981a91b/i2c.cpp#L5
> 
> Generally, it's best practice to separate out the construction of a variable/allocation of memory and the actual "beginning" of the device by sending some data to it over a serial protocol. Not super major just something to note

Makes sense. That would just mean, in the i2c case at least, creating a seperate "open" function
